### ISSUE TITLE
Unable to deploy Kafka when createGlobalResources is set to false

### ISSUE THREAD

**Issue by mohitruhaan** [strimzi/strimzi-kafka-operator#6287]
Hello Experts,

While trying to deploy Strimzi version 0.27, we tried to set the flag: createGlobalResources to "false" in Values.yaml.
To our understanding, it means strimzi should work fine when used for managing Kafka within a namespace, this not needing global resources.
But we are not able to deploy Kafka within a namespace. We observe service account and authorization related errors as below:

```
2022-02-01 10:55:53 ERROR AbstractOperator:432 - Reconciliation #2(watch) Kafka(databus/databus-cluster): Reconciliation completion handler failed
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://10.96.0.1/apis/kafka.strimzi.io/v1beta2/namespaces/databus/kafkas/databus-cluster. Message: Forbidden!Configured _**service account doesn't have access. Service account may have been revoked.**_ kafkas.kafka.strimzi.io "databus-cluster" is forbidden: User "system:serviceaccount:databus:strimzi-cluster-operator" cannot get resource "kafkas" in API group "kafka.strimzi.io" in the namespace "databus": **_RBAC: [clusterrole.rbac.authorization.k8s.io "strimzi-cluster-operator-namespaced" not found, clusterrole.rbac.authorization.k8s.io "strimzi-entity-operator" not found]._**
```

Could you please suggest? Thanks!

**Comment by scholzj** [comment]
> To our understanding, it means strimzi should work fine when used for managing Kafka within a namespace, this not needing global resources.

That is not correct. IIRC, what this mode does is that it will not install the global resources such as the ClusterRoles or ClusterRoleBindings. But it expects you to do so outside of the Helm Installation (at least for the ClusterRoles and CRDs). See #1436 for more details.

You can in theory not install the ClusterRoleBindings and have some basic functionality. But there will be some limitations to what the operator can do (no rack awareness, storage resizing, node port listeners). You would also still need to install the ClusterRoles for it.
